impeller/renderer/context.h

```diff
@@ -154,6 +154,20 @@ class Context {
   ///
   virtual std::shared_ptr<CommandBuffer> CreateCommandBuffer() const = 0;
 
+  //---------------------------------------------------------------------------
+  /// @brief Indicate the beginning of a presentation cycle.
+  ///
+  ///        This may or may not indicate a new frame, or there may be multiple
+  ///        presents within a single frame due to platform view presence.
+  virtual void DidAcquireSurfaceFrame() const = 0;
+
+  //---------------------------------------------------------------------------
+  /// @brief Indicate the ending of a presentation cycle.
+  ///
+  ///        This may or may not indicate a new frame, or there may be multiple
+  ///        presents within a single frame due to platform view presence.
+  virtual void DidFinishSurfaceFrame() const = 0;
```

Comment:
Should we call these `BeginFrame()` and `EndFrame()`, since surfaces aren't necessarily involved here?

We should also document that GPU resources may leak if they aren't called properly.